src/builders/signer.builders.ts

```diff
@@ -0,0 +1,111 @@
+import {encodeIcrcAccount, type IcrcTransferArg} from '@dfinity/ledger-icrc';
+import {
+  arrayOfNumberToUint8Array,
+  fromNullable,
+  isNullish,
+  nonNullish,
+  uint8ArrayToHexString
+} from '@dfinity/utils';
+import {TransferArgs} from '../constants/icrc.idl.constants';
+import type {icrc21_consent_info} from '../declarations/icrc-21';
+import type {SignerBuilderFn, SignerBuildersResult} from '../types/signer-builders';
+import {formatAmount} from '../utils/format.utils';
+import {decodeIdl} from '../utils/idl.utils';
+
+/**
+ * Builds a content message for an ICRC-1 transfer by decoding the arguments for a potential call.
+ * This is used as a workaround when the targeted canister does not comply with the ICRC-21 standard — i.e. it has not implemented the related endpoints.
+ *
+ * The implementation is similar to the Markdown generated by the ICRC ledger implementation.
+ * @link https://github.com/dfinity/ic/blob/master/packages/icrc-ledger-types/src/icrc21/lib.rs#L153
+ *
+ * @param {Object} params - Parameters for building the consent message.
+ * @param {Uint8Array} params.arg - Encoded arguments for the ICRC-1 transfer.
+ * @param {Principal} params.owner - Principal ID of the sender (owner) account.
+ * @param {Object} params.token - Token metadata including symbol, decimals, and fee.
+ * @param {string} params.token.symbol - The symbol of the token.
+ * @param {number} params.token.decimals - The number of decimals for the token.
+ * @param {bigint} params.token.fee - Default fee for the token transfer.
+ * @returns {Promise<SignerBuildersResult>} - A result containing either the consent message or an error.
+ *
+ **/
+export const buildContentMessageIcrc1Transfer: SignerBuilderFn = async ({
+  arg,
+  owner,
+  token: {symbol: tokenSymbol, decimals: tokenDecimals, fee: tokenFee}
+}): Promise<SignerBuildersResult> => {
+  try {
+    const {
+      amount,
+      from_subaccount: fromSubaccount,
+      to: {owner: toOwner, subaccount: toSubaccount},
+      fee,
+      memo
+    } = decodeIdl<IcrcTransferArg>({
+      recordClass: TransferArgs,
+      bytes: arg
+    });
+
+    // eslint-disable-next-line import/no-relative-parent-imports
+    const {default: en} = await import('../i18n/en.json');
```

Comment:
Does it make sense to add a TODO for posterity? Either way, we will have to look for all   usages when it will be internationalised

Reply:
Good idea 👍

I created a separate PR #373 to provide the TODOs because we can add two of those.